### DISCUSSION TITLE
Unrecognized pair style 'deepmd'

### DISCUSSION THREAD

**Discussion by alundilong** [deepmodeling/deepmd-kit#3092] [Q&A]
### **run 'lmp -i input.lammps' shows**

LAMMPS (2 Aug 2023 - Update 2)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
Reading data file ...
  orthogonal box = (0.224452 0.231356 -0.157463) to (37.82585 37.832754 37.443935)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  900 atoms
  read_data CPU = 0.002 seconds
ERROR: Unrecognized pair style 'deepmd' (src/force.cpp:271)
Last command: pair_style      deepmd graph_compressed.pb

### **'lmp -h' shows the following, deepmd does NOT show up**

* Pair styles:

born            buck            buck/coul/cut   coul/cut        coul/debye      
coul/dsf        coul/wolf       reax            mesont/tpm      hybrid          
hybrid/overlay  hybrid/scaled   lj/cut          lj/cut/coul/cut lj/expand       
morse           soft            table           yukawa          zbl             
zero            

* Bond styles:

### **However, Installation does indicate deepmd-kit is successfully compiled**

[ 96%] Building CXX object CMakeFiles/lammps.dir/usr/Downloads/lammps-2Aug2023/src/write_coeff.cpp.o
[ 96%] Building CXX object CMakeFiles/lammps.dir/usr/Downloads/lammps-2Aug2023/src/write_data.cpp.o
[ 96%] Building CXX object CMakeFiles/lammps.dir/usr/Downloads/lammps-2Aug2023/src/write_dump.cpp.o
[ 97%] Building CXX object CMakeFiles/lammps.dir/usr/Downloads/lammps-2Aug2023/src/write_restart.cpp.o
[ 97%] Building CXX object CMakeFiles/lammps.dir/usr/Downloads/lammps-deepmd-package/deepmd-kit/source/lmp/compute_deeptensor_atom.cpp.o
[ 97%] Building CXX object CMakeFiles/lammps.dir/usr/Downloads/lammps-deepmd-package/deepmd-kit/source/lmp/fix_dplr.cpp.o
**[ 97%] Building CXX object CMakeFiles/lammps.dir/usr/Downloads/lammps-deepmd-package/deepmd-kit/source/lmp/pair_deepmd.cpp.o**
[ 97%] Building CXX object CMakeFiles/lammps.dir/usr/Downloads/lammps-deepmd-package/deepmd-kit/source/lmp/pppm_dplr.cpp.o
[ 98%] Building CXX object CMakeFiles/lammps.dir/usr/Downloads/lammps-2Aug2023/src/KSPACE/pppm.cpp.o
[ 98%] Building CXX object CMakeFiles/lammps.dir/usr/Downloads/lammps-2Aug2023/src/KSPACE/fft3d.cpp.o
[ 98%] Building CXX object CMakeFiles/lammps.dir/usr/Downloads/lammps-2Aug2023/src/KSPACE/fft3d_wrap.cpp.o
[ 98%] Building CXX object CMakeFiles/lammps.dir/usr/Downloads/lammps-2Aug2023/src/KSPACE/remap.cpp.o
[100%] Building CXX object CMakeFiles/lammps.dir/usr/Downloads/lammps-2Aug2023/src/KSPACE/remap_wrap.cpp.o
[100%] Building CXX object CMakeFiles/lammps.dir/usr/Downloads/lammps-2Aug2023/src/EXTRA-FIX/fix_ttm.cpp.o
[100%] Linking CXX shared library liblammps.so

**I have seen other users have reported similar issue, but I believe mine is different from them.
Btw, I am running the code on Desktop, not on cluster. Thank you for your attention.**

**Comment by alundilong** [comment]
if I use 'ldd lmp' to find which so files are linked,

	linux-vdso.so.1 (0x00007ffd83db3000)
	liblammps.so.0 => /usr/Downloads/lammps-2Aug2023/build/liblammps.so.0 (0x00007fef1f200000)
	libmpich.so.12 => /lib/x86_64-linux-gnu/libmpich.so.12 (0x00007fef1cbfc000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fef1c9d0000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fef1f955000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fef1c7a8000)
	libdeepmd_c.so => /usr/Downloads/lammps-deepmd-package/installation/lib/libdeepmd_c.so (0x00007fef1f937000)
	libgomp.so.1 => /lib/x86_64-linux-gnu/libgomp.so.1 (0x00007fef1f8ed000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fef1c6c1000)
	libhwloc.so.15 => /lib/x86_64-linux-gnu/libhwloc.so.15 (0x00007fef1c665000)
	libfabric.so.1 => /lib/x86_64-linux-gnu/libfabric.so.1 (0x00007fef1c510000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fef1f982000)
	libdeepmd_cc.so => /usr/Downloads/lammps-deepmd-package/installation/lib/libdeepmd_cc.so (0x00007fef1c498000)
	libudev.so.1 => /lib/x86_64-linux-gnu/libudev.so.1 (0x00007fef1c46e000)
	librdmacm.so.1 => /lib/x86_64-linux-gnu/librdmacm.so.1 (0x00007fef1c44f000)
	libefa.so.1 => /lib/x86_64-linux-gnu/libefa.so.1 (0x00007fef1f8de000)
	libibverbs.so.1 => /lib/x86_64-linux-gnu/libibverbs.so.1 (0x00007fef1c42c000)
	libpsm_infinipath.so.1 => /lib/x86_64-linux-gnu/libpsm_infinipath.so.1 (0x00007fef1c3d4000)
	libpsm2.so.2 => /lib/x86_64-linux-gnu/libpsm2.so.2 (0x00007fef1c366000)
	libtensorflow_cc.so.2 => /usr/local/lib/python3.10/dist-packages/tensorflow/libtensorflow_cc.so.2 (0x00007fef03900000)
	libtensorflow_framework.so.2 => /usr/local/lib/python3.10/dist-packages/tensorflow/libtensorflow_framework.so.2 (0x00007fef016a6000)
	libnl-3.so.200 => /lib/x86_64-linux-gnu/libnl-3.so.200 (0x00007fef01683000)
	libnl-route-3.so.200 => /lib/x86_64-linux-gnu/libnl-route-3.so.200 (0x00007fef01600000)
	libinfinipath.so.4 => /lib/x86_64-linux-gnu/libinfinipath.so.4 (0x00007fef015ed000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fef015e8000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fef015e3000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fef015de000)
	libuuid.so.1 => /lib/x86_64-linux-gnu/libuuid.so.1 (0x00007fef015d5000)
	libnuma.so.1 => /lib/x86_64-linux-gnu/libnuma.so.1 (0x00007fef015c8000)

**Comment by njzjz** [comment]
Thanks for the report. #3097 should fix it.
